backend/api/simulation.py

```
@@ -3218,74 +3230,35 @@ async def get_simulation_grading(
-                    result = pyjson.loads(json_str)
-                else:
-                    result = pyjson.loads(raw_content)
-                score = int(result.get("score", 0))
-                feedback = result.get("feedback", "No feedback provided.")
+                
+                score = grading_result.get("score", 0)
+                feedback = grading_result.get("feedback", "No feedback provided.")
+                
+                print(f"[DEBUG] RAG grading completed for scene '{scene.title}': score={score}")
+                
             except Exception as e:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)